automated/linux/mmtests/mmtests.sh

```diff
@@ -146,75 +138,105 @@ prepare_system() {
-  export NUMLLCS
+  KERNEL_VERSION=$(uname -r)
+  cat <<EOF
+{
+  "MEMTOTAL_BYTES": "${MEMTOTAL_BYTES:-}",
+  "NUMCPUS": "${NUMCPUS:-}",
+  "NUMNODES": "${NUMNODES:-}",
+  "LLC_INDEX": "${LLC_INDEX:-}",
+  "NUMLLCS": "${NUMLLCS:-}",
+  "KERNEL_VERSION": "${KERNEL_VERSION:-}",
+  "MMTESTS_ITERATIONS": "${MMTESTS_ITERATIONS:-}",
+  "MMTESTS_CONFIG_FILE": "${MMTESTS_CONFIG_FILE:-}"
+}
+EOF
+}
 
-  chmod u+x ./"${MMTESTS_CONFIG_FILE}"
-  eval 'source ./${MMTESTS_CONFIG_FILE}'
+collect_results() {
+  # Collect benchmark run details in JSON object.
+  details=$(collect_details)
+  # Dump details to temp file
+  details_file=$(mktemp)
+  echo "$details" > "$details_file"
   # Extract results data from available logs for each benchmark in JSON format.
-  # JSON files will be available in mmtests root directory.
-
-  # Note: benchmark name is not always equal to benchmark name from config file.
-  if [ "${MMTESTS_TYPE_NAME}" != "${MMTESTS}" ]; then
-    EXTRACT_NAMES="${MMTESTS}"
-  else
-    EXTRACT_NAMES="${MMTESTS_TYPE_NAME}"
-  fi
-
-  echo "test(s) to extract: ${EXTRACT_NAMES}"
-  for benchmark_name in ${EXTRACT_NAMES}; do
-    echo "results for: $benchmark_name"
-    ${MMTEST_EXTR} -d work/log/ -b "${benchmark_name}" -n benchmark --print-json >> "../${MMTESTS_TYPE_NAME}_${benchmark_name}.json"
-
-    altreports=${altreport_mappings[${benchmark_name}]}
-    for altreport in ${altreports}; do
-      ${MMTEST_EXTR} -d work/log/ -b "${benchmark_name}" -n benchmark \
-      -a "${altreport}" --print-json > "../${MMTESTS_TYPE_NAME}_${benchmark_name}${altreport}.json"
-    done
-  done
-
-  env_variables_prefix=${env_variable_mappings[${MMTESTS_TYPE_NAME}]}
-  if [ -z "$env_variables_prefix" ]; then
-    env_variables_prefix=${MMTESTS_TYPE_NAME^^}
-  fi
-
-  vars=""
-  eval 'vars=${!'"$env_variables_prefix"'*}'
-  for variable in ${vars}; do
-    mykey=CONFIG_${variable}
-    mykey=${mykey//_/-}
-    myvalue=${!variable}
-    echo "$mykey":"$myvalue"
-    tmp=$(mktemp)
-    jq -c --arg k "$mykey" --arg v "$myvalue" '. += {($k):$v}' \
-      "../${MMTESTS_TYPE_NAME}_${benchmark_name}.json" > "$tmp" \
-      || { echo "jq operation failed"; exit 1; }
-    mv "$tmp" "../${MMTESTS_TYPE_NAME}_${benchmark_name}.json"
+  mapfile -t jsons < <(extract_json)
+  for json in "${jsons[@]}"; do
+    # Create a temp file to hold the merged JSON
+    merge_file=$(mktemp)
+    # Merge details and results JSON
+    jq -n \
+        --argfile d "$details_file" \
+        --argfile r "$json" \
+        '{details: $d, results: $r}' > "$merge_file"
+    # Replace results file
+    mv "$merge_file" "$json"
   done
-
-  chmod a+r "../$MMTESTS_TYPE_NAME"*".json"
-  popd || exit
 }
 
 ! check_root && error_msg "Please run this script as root."
 
 if [ "${SKIP_INSTALL}" = "true" ]; then
-  info_msg "${MMTESTS_TYPE_NAME} installation skipped"
+  info_msg "Installation skipped"
```

Comment:
not sure if this change belongs in this patch.

Reply:
MMTESTS_TYPE_NAME was removed in this patch.